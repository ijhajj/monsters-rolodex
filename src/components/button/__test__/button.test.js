import React, { createElement } from 'react';
//import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import Button from '../button';
import '@testing-library/jest-dom/extend-expect';
import renderer from "react-test-renderer";

test("Renders without crashing!!!",()=>{
    const div = createElement("div")
    render(<Button></Button>,div)
})

test("Renders button correctly",()=>{
    const {getByTestId} = render(<Button label="Click Testing"/>)
    expect(getByTestId('button')).toHaveTextContent("Click Testing")
})
test("Renders button correctly",()=>{
    const {getByTestId} = render(<Button label="Failing Button"/>)
    expect(getByTestId('button')).toHaveTextContent("Failing Button")
})
test("Matches Snapshot",()=>{
    const jsonTree = renderer.create(<Button label="Snapshot Check"/>).toJSON();
    expect(jsonTree).toMatchSnapshot();
})