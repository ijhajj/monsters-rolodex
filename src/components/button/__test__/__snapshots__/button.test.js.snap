// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  Snapshot Check
</div>
`;
